Library/Homebrew/os/mac/sdk.rb

```diff
@@ -93,10 +93,21 @@ def sdk_paths
 
             # Use unversioned SDK path on Big Sur to avoid issues such as:
             # https://github.com/Homebrew/homebrew-core/issues/67075
-            if OS::Mac.version >= :big_sur
-              sdk_path = File.join(sdk_prefix, "MacOSX.sdk")
-              version = OS::Mac.full_version
-              paths[version] = sdk_path if File.directory?(sdk_path)
+            sdk_path = File.join(sdk_prefix, "MacOSX.sdk")
+            if OS::Mac.version >= :big_sur && File.directory?(sdk_path)
+              sdk_settings_json = File.join(sdk_path, "SDKSettings.json")
+              sdk_settings_plist = File.join(sdk_path, "SDKSettings.plist")
+
+              if File.exist?(sdk_settings_json)
+                version = JSON.parse(File.read(sdk_settings_json))["Version"]
+              elsif File.exist?(sdk_settings_plist)
+                p = Utils.popen_read("plutil", "-convert", "json", "-o", "-", sdk_settings_plist)
```

Comment:
`p` is a built-in function so would be nice to use something a bit longer.

Reply:
Also `safe_popen_read`.

Reply:
Think this can be skipped entirely in favour of just the JSON.

Reply:
I actually intentionally chose `popen_read` because I don't want it to error out if it fails.